NModbus/Interfaces/IModbusMaster.cs

```diff
@@ -58,6 +58,15 @@ public interface IModbusMaster : IDisposable
         /// <returns>Holding registers status.</returns>
         ushort[] ReadHoldingRegisters(byte slaveAddress, ushort startAddress, ushort numberOfPoints);
 
+        /// <summary>
+        ///    Reads contiguous block of holding registers with 32 bit register size.
+        /// </summary>
+        /// <param name="slaveAddress">Address of device to read values from.</param>
+        /// <param name="startAddress">Address to begin reading.</param>
+        /// <param name="numberOfPoints">Number of holding registers to read.</param>
+        /// <returns>Holding registers status.</returns>
+        uint[] ReadHoldingRegisters32(byte slaveAddress, ushort startAddress, ushort numberOfPoints);

```

Comment:
Rather than alter the `IModbusMaster` interface, I'm wondering if it would be more powerful / flexible / less change to the most used interface if we moved this logic to extension methods.

Maybe something like this:

```c#
public static class IModbusMasterExtensions
{
   public static uint[] ReadHoldingRegisters32(this IModbusMaster master, byte slaveAddress, ushort startAddress, ushort numberOfPoints)
  {
     var holdingRegisters = master.ReadHoldingRegisters(slaveAddress,, startAddress, numberOfPoints * 2);

      //TODO: Math to assemble the ushort[] array into uint[]
  }
}
```

Reply:
I only have corrected existing functions. Discussions about architecture needs to be done by main developers.

Reply:
It looks like you moved the functionality from the old extension methods in `EnronModbus` to `IModbusMaster`. Was it not possible to fix those extension methods in place?

Reply:
Yes thats a valid concern. I have moved the functions to the enron modbus class.